crates/worker/src/partition_processor_manager/mod.rs

```diff
@@ -656,7 +696,7 @@ impl PartitionProcessorManager {
         }
     }
 
-    fn request_partition_snapshots(&mut self) {
+    fn trigger_periodic_partition_snapshots(&mut self) {
```

Comment:
Why aren't we directly calling `spawn_create_snapshot_task` here? Sending ourselves a new message which might fail if the channel is full seems a bit unnecessary.

Reply:
Where are we checking that only leaders create snapshots?

Reply:
I had missed this in this in the initial revision, added now! (Line `748`.)

Reply:
None whatsoever! It was on my list but I clearly dropped it 😅